python/src/gz/sim/TestFixture.cc

```diff
@@ -56,7 +56,14 @@ defineSimTestFixture(pybind11::object module)
       [](TestFixture* self, std::function<void(
           const UpdateInfo &, EntityComponentManager &)> _cb)
       {
+        // Add explicit scoped acquire and release of GIL, so that Python Systems
+        // can be executed
+        // This acquire and release is only required from the PythonSystem code
+        // However, adding this here may prevent undefined or unintended behaviors
+        // in future
+        pybind11::gil_scoped_acquire gil;
         self->OnPreUpdate(_cb);
+        pybind11::gil_scoped_release gilr;
```

Comment:
Is this necessary? I thought the scoped lock above would automatically release when going out of scope